File: src/types/Note.ts
export type Note = {
    id: string;
    noteDate: Date;
    note: string;
};

export type NoteDTO = {
    id: string;
    note_date: string;
    note: string;
}

export type CreateNotePayload = {
    profileId: string;
    note: string;
    noteDate: Date;
}

export type CreateNoteInput = {
    profile_id: string;
    note: string;
    note_date: string;
}

export type EditNotePayload = {
    profileId: string;
    noteId: string;
    note: string;
}

export type EditNoteInput = {
    profile_id: string;
    note_id: string;
    note: string;
}   

export type ListNotesPayload = {
    profileId: string;
}

export type ListNotesInput = {
    profile_id: string;
}

File: src-tauri/src/models/note.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Deserialize)]
pub struct CreateNoteInput {
    pub profile_id: String,
    pub note_date: String,
    pub note: String,
}

#[derive(Debug, Deserialize)]
pub struct EditNoteInput {
    pub profile_id: String,
    pub note_id: String,
    pub note: String,
}

#[derive(Debug, Deserialize)]
pub struct ListNotesInput {
    pub profile_id: String,
}

#[derive(Debug, Serialize)]
pub struct Note {
    pub id: String,
    pub profile_id: String,
    pub note_date: String,
    pub note: String,
    pub created_at: String,
    pub updated_at: Option<String>,
}
